lib/Serialization/Deserialization.cpp

```diff
@@ -283,7 +307,15 @@ Expected<Pattern *> ModuleFile::readPattern(DeclContext *owningDC) {
   SmallVector<uint64_t, 8> scratch;
 
   BCOffsetRAII restoreOffset(DeclTypeCursor);
-  auto next = DeclTypeCursor.advance(AF_DontPopBlockAtEnd);
+  Expected<llvm::BitstreamEntry> maybeNext =
+      DeclTypeCursor.advance(AF_DontPopBlockAtEnd);
+  if (!maybeNext) {
+    // FIXME this drops the error diagnostic on the floor.
+    consumeError(maybeNext.takeError());
+    error();
```

Comment:
`fatal` again, throughout this whole file wherever you would be calling `error()`. Besides what I mentioned above, it's `noreturn`, so you don't have to make a fake return value.

Reply:
I've updated these.